File: go-solutions/1004_longest_ones.go
package leetcode

func LongestOnes(nums []int, k int) int {
	l, r := 0, 0
	for r = 0; r < len(nums); r++ {
		if nums[r] == 0 {
			k--
		}
		if k < 0 {
			k += 1 - nums[l]
			l++
		}
	}
	return r - l
}

File: ts-solutions/src/1004-longest-ones.ts
export function longestOnes(nums: number[], k: number): number {
	let left = 0
	let right = 0
	for (; right < nums.length; right++) {
		if (nums[right] === 0) {
			k--
		}
		if (k < 0) {
			k += 1 - nums[left]
			left++
		}
	}
	return right - left
}
